application/src/main/java/org/togetherjava/tjbot/commands/moderation/WhoIsCommand.java

```diff
@@ -0,0 +1,263 @@
+            .setDescription(descriptionBuilder);
+
+
+        return event.replyEmbeds(embedBuilder.build())
+            .addActionRow(Button.of(ButtonStyle.LINK, "discord://-/users/" + user.getId(),
+                    "Click to see profile"));
+    }
+
+    private static @NotNull String handleVoiceState(@NotNull final Member member) {
+        GuildVoiceState voiceState = Objects.requireNonNull(member.getVoiceState(),
+                "The given voiceState cannot be null");
+        if (voiceState.inVoiceChannel()) {
+            return "\n**In voicechannel:** " + (voiceState.getChannel().getAsMention());
+        } else {
+            return "";
+        }
+    }
+
+
+    /**
+     * Generates whois embed based on the given parameters.
+     *
+     * @param event the {@link SlashCommandEvent}
+     * @param user the {@link User} getting whois'd
+     * @param profile the {@link net.dv8tion.jda.api.entities.User.Profile} of the whois'd user
+     * @param effectiveColor the {@link Color} that the embed will become
+     * @return the generated {@link EmbedBuilder}
+     */
+    private static @NotNull EmbedBuilder generateEmbedBuilder(
+            @NotNull final SlashCommandEvent event, @NotNull final User user,
+            final @NotNull User.Profile profile, final Color effectiveColor) {
+
+
+        EmbedBuilder embedBuilder = new EmbedBuilder().setThumbnail(user.getEffectiveAvatarUrl())
+            .setColor(effectiveColor)
+            .setFooter("Requested by " + event.getUser().getAsTag(),
+                    event.getMember().getEffectiveAvatarUrl())
+            .setTimestamp(Instant.now());
+
+        if (null != profile.getBannerId()) {
+            embedBuilder.setImage(profile.getBannerUrl() + "?size=4096");
+        }
+
+        return embedBuilder;
+    }
+
+    /**
+     * Handles boosting properties of a {@link Member}
+     *
+     * @param member the {@link Member} to take the booster properties from
+     * @return user readable {@link String}
+     */
+    private static @NotNull String handlePossibleBooster(final @NotNull Member member) {
+        OffsetDateTime timeBoosted = member.getTimeBoosted();
+        if (null != timeBoosted) {
+            return "\n**Is booster:** true \n**Boosting since:** "
+                    + DATE_TIME_FORMAT.format(timeBoosted);
+        } else {
+            return "\n**Is booster:** false";
+        }
+    }
+
+    /**
+     * Handles the user's identifying properties (such as ID, tag)
+     *
+     * @param user the {@link User} to take the identifiers from
+     * @return user readable {@link StringBuilder}
+     */
+    private static @NotNull StringBuilder handleUserIdentification(final @NotNull User user) {
+        return new StringBuilder("**Mention:** ").append(user.getAsMention())
+            .append("\n**Tag:** ")
+            .append(user.getAsTag())
+            .append("\n**ID:** ")
+            .append(user.getId());
+    }
```

Comment:
same issue with the name i also had with the voice state. a "handle" method is, for me, a method that actually does `event.queue()` and similar stuff, i.e. without a real return value.

this is more like a "format/tostring the current state of property x".

Reply:
Format feels wrong to me when comparing to normal usages of the format method.
`userIdentificationToString` is acceptable, but still feels weird?

Do you've any other suggestions? Otherwise I'll go with `userIdentificationToString`